daemon/graphdriver/overlay2/overlay.go

```diff
@@ -204,7 +205,16 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		logger.Warnf("Unable to detect whether overlay kernel module supports index parameter: %s", err)
 	}
 
-	logger.Debugf("backingFs=%s, projectQuotaSupported=%v, indexOff=%q", backingFs, projectQuotaSupported, indexOff)
+	needsUserXattr, err := overlayutils.NeedsUserXAttr(home)
+	if err != nil {
+		logger.Warnf("Unable to detect whether overlay kernel module needs \"userxattr\" parameter: %s", err)
+	}
+	if needsUserXattr {
+		userxattr = "userxattr,"
+	}
+
+	logger.Debugf("backingFs=%s, projectQuotaSupported=%v, indexOff=%q, userxattr=%q",
```

Comment:
Can we remove this from the `init`?

Reply:
This Debugf is not new in this PR.

Reply:
I'm personally ok with keeping this (for now) as it's not new indeed.

Looking at the code, I also want to do some refactoring after this is merged; the string-concatting code is becoming a  bit messy and, assuming that won't affect performance, I'd like to use a `[]string` for these options, and `strings.Join()` them at the end (I think that will make the code more readable).

Reply:
Oh this was a github UI issue, its not in the init like I thought.